camel-idea-plugin/src/main/java/org/apache/camel/idea/annotator/CamelAnnotator.java

```diff
@@ -0,0 +1,220 @@
+    private void validateSimpleExpression(@NotNull PsiElement element, @NotNull AnnotationHolder holder, String fromElement) {
+        if (fromElement.contains("${") &&  (IdeaUtils.isCamelRouteSimpleExpression(element))) {
+            CamelCatalog catalogService = ServiceManager.getService(element.getProject(), CamelCatalogService.class).get();
+            SimpleValidationResult validateSimpleResult = catalogService.validateSimpleExpression(fromElement);
+            if (!validateSimpleResult.isSuccess()) {
+                String error = validateSimpleResult.getError();
+                int propertyIdx = fromElement.indexOf("simple");
+                int propertyLength = fromElement.length() + 8;
+                TextRange range = new TextRange(element.getTextRange().getStartOffset() + propertyIdx,
+                    element.getTextRange().getStartOffset() + propertyIdx + propertyLength);
+                holder.createErrorAnnotation(range, error);
+            }
+
+        }
+    }
+
+    private void extractMapValue(EndpointValidationResult validateEndpointProperties, Map<String, String> validationMap,
+                                 String fromElement, @NotNull PsiElement element, @NotNull AnnotationHolder holder) {
+        if ((!validateEndpointProperties.isSuccess()) && validationMap != null) {
+            String propertyValue = validationMap.values().iterator().next();
+            String propertyKey = validationMap.keySet().iterator().next();
+            int propertyIdx = propertyValue.isEmpty() ? fromElement.indexOf(propertyKey) + 1 : fromElement.indexOf(propertyValue) + 1;
+            int propertyLength = propertyValue.isEmpty() ? propertyKey.length() : propertyValue.length();
+            TextRange range = new TextRange(element.getTextRange().getStartOffset() + propertyIdx,
+                element.getTextRange().getStartOffset() + propertyIdx + propertyLength);
+            holder.createErrorAnnotation(range, summaryErrorMessage(validateEndpointProperties));
+        }
+    }
+
+    /**
+     * A human readable summary of the validation errors.
+     *
+     * @return the summary, or <tt>null</tt> if no validation errors
+     */
+    private String summaryErrorMessage(EndpointValidationResult result) {
+
+        if (result.getIncapable() != null) {
+            return "\tIncapable of parsing uri: " + result.getIncapable();
+        } else if (result.getSyntaxError() != null) {
+            return "\tSyntax error: " + result.getSyntaxError();
+        } else if (result.getUnknownComponent() != null) {
+            return "\tUnknown component: " + result.getUnknownComponent();
+        }
+
+        // for each invalid option build a reason message
+        Map<String, String> options = new LinkedHashMap<>();
+        if (result.getUnknown() != null) {
+            for (String name : result.getUnknown()) {
+                if (result.getUnknownSuggestions() != null && result.getUnknownSuggestions().containsKey(name)) {
```

Comment:
To enable suggestions we need to add camel-catalog-lucene to the dependency, and turn it on at the camel-catalog service. 

See this test: https://github.com/apache/camel/blob/master/platforms/catalog-lucene/src/test/java/org/apache/camel/catalog/lucene/CamelCatalogLuceneTest.java

And just food for thought. Maybe those suggestions can somehow be in IDEA shown as a quick-fix or drowdown list of possible corrections. Just something to work on in the future, if that can be possible.